Deploy/modules/PrivateDNSZonesDeploy.bicep

```
@@ -0,0 +1,201 @@
+
+param vNetID string
+param vNetName string
+
+param ctrlDeployPurview bool
+param ctrlDeployAI bool
+param ctrlDeployStreaming bool
+//param crtlDeployDataShare bool
+
+
+var environmentStorageDNS = environment().suffixes.storage
+
+//Private DNS Zones required for Storage DFS Private Link: privatelink.dfs.core.windows.net
+//Required for Azure Data Lake Gen2
+module m_privateDNSZoneStorageDFS './PrivateDNSZone.bicep' = {
+  name: 'PrivateDNSZoneStorageDFS'
+  params: {
+    dnsZoneName: 'privatelink.dfs.${environmentStorageDNS}'
+    vNetID: vNetID
+    vNetName: vNetName
+  }
+}
+
+//Private DNS Zones required for Storage Blob Private Link: privatelink.blob.core.windows.net
+//Required for Purview, Azure ML
+module m_privateDNSZoneStorageBlob 'PrivateDNSZone.bicep' = if (ctrlDeployPurview == true || ctrlDeployAI == true){
+  name: 'PrivateDNSZoneStorageBlob'
+  params: {
+    dnsZoneName: 'privatelink.blob.${environmentStorageDNS}'
+    vNetID: vNetID
+    vNetName: vNetName
+  }
+}
+
+//Private DNS Zones required for Storage Queue Private Link: privatelink.queue.core.windows.net
+//Required for Purview
+module m_privateDNSZoneStorageQueue 'PrivateDNSZone.bicep' = if (ctrlDeployPurview == true) {
+  name: 'PrivateDNSZoneStorageQueue'
+  params: {
+    dnsZoneName: 'privatelink.queue.${environmentStorageDNS}'
+    vNetID: vNetID
+    vNetName: vNetName
+  }
+}
+
+//Private DNS Zones required for Storage File Private Link: privatelink.queue.core.windows.net
+//Required for Azure ML Storage Account
+module m_privateDNSZoneStorageFile 'PrivateDNSZone.bicep' = if (ctrlDeployAI == true) {
+  name: 'PrivateDNSZoneStorageFile'
+  params: {
+    dnsZoneName: 'privatelink.file.${environmentStorageDNS}'
+    vNetID: vNetID
+    vNetName: vNetName
+  }
+}
```

Comment says queue, but zone is for files.
```suggestion
//Private DNS Zones required for Storage File Private Link: privatelink.file.core.windows.net
//Required for Azure ML Storage Account
module m_privateDNSZoneStorageFile 'PrivateDNSZone.bicep' = if (ctrlDeployAI == true) {
  name: 'PrivateDNSZoneStorageFile'
  params: {
    dnsZoneName: 'privatelink.file.${environmentStorageDNS}'
    vNetID: vNetID
    vNetName: vNetName
  }
}
```

---

Done